CsharpExtrasTest/OneBased/OneBasedArray2DTest.cs

```diff
@@ -97,12 +99,79 @@ public void Given_1dArray_When_ConvertedTo2dArrayAcrossRows_Then_New2dArrayHasCo
+        {
+            OneBasedArray2DImpl<int> oneBasedData = new OneBasedArray2DImpl<int>(data);
+            OneBasedArrayImpl<int> oneBasedDataToWrite = new OneBasedArrayImpl<int>(dataToWrite);
+            IOneBasedArray<int> oneBasedExpected = new OneBasedArrayImpl<int>(expected);
+            oneBasedData.WriteToColumn(oneBasedDataToWrite, column, offset);
+            for (int i = 1; i <= oneBasedExpected.Length; i++)
+            {
+                Assert.AreEqual(oneBasedExpected[i], oneBasedData[i, column],
+                    string.Format("Mismatch at row {0} column {1} for offset {2}", i, column, offset));
+            }
+        }
+
+        #region Test Data
+        private static IEnumerable<object[]> ProviderForWrite1DArrayTo2DColumn()
+        {
+            return new List<object[]> {
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] {21, 22, 23}, 2, 1,
+                    new int[]{11, 21, 22, 23}},
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] {21, 22, 23}, 2, 0,
+                    new int[]{21, 22, 23, 14}},
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] { 1, 2, 3, 4 }, 1, 0,
+                    new int[]{1, 2, 3, 4}},
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] { 1, 2, 3, 4 }, 1, 2,
+                    new int[]{1, 2, 1, 2}},
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] { 21, 22, 23, 24 }, 1, 4,
+                    new int[]{1, 2, 3, 4}},
+                new object[] { new int[,] { { 1, 11 }, { 2, 12 }, { 3, 13 }, { 4, 14 } }, new int[] { 21, 22, 23, 24 }, 1, -1,
+                    new int[]{22, 23, 24, 4}},
+            };
+        }
+
+        private static IEnumerable<object[]> ProviderForWrite1DArrayTo2DRow()
+        {
+            return new List<object[]> {
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] {21, 22, 23}, 2, 1,
+                    new int[]{11, 21, 22, 23}},
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] {21, 22, 23}, 2, 0,
+                    new int[]{21, 22, 23, 14}},
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] { 1, 2, 3, 4 }, 1, 0,
+                    new int[]{1, 2, 3, 4}},
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] { 1, 2, 3, 4 }, 1, 2,
+                    new int[]{1, 2, 1, 2}},
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] { 21, 22, 23, 24 }, 1, 4,
+                    new int[]{1, 2, 3, 4}},
+                new object[] { new int[,] { { 1, 2, 3, 4 }, {11, 12, 13, 14} }, new int[] { 21, 22, 23, 24 }, 1, -1,
+                    new int[]{22, 23, 24, 4}},
+            };
+        }

```

Comment:
Reduce the number of cases since this is already tested in the zero-based functionality